syntax = "proto2";

option optimize_for = LITE_RUNTIME;

message VideoMessage{

  enum Action {
    START = 0;
    STOP = 1;
  };

  required Action action = 1;
  required string pipeline = 2;

  required int32 device_id = 3;
  required int32 width = 4;
  required int32 height = 5;
  required int32 framerate_numerator = 6;
  required int32 framerate_denumenator = 7;

}